code/src/App.js

```diff
@@ -1,9 +1,83 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react';
+
+import { HAPPY_THOUGHTS_URL, HEARTS_URL } from './reusable/urls';
+
+import MessageForm from 'components/MessageForm';
+import MessageList from 'components/MessageList';
 
 export const App = () => {
+  const [messageList, setMessageList] = useState([]);
+  const [messageNew, setMessageNew] = useState(['']);
+
+  useEffect(() => {
+    fetchMessageList();
+  }, []);
+
+  const fetchMessageList = () => {
+    fetch(HAPPY_THOUGHTS_URL)
+      .then(res => res.json())
+      .then(inputs => setMessageList(inputs))
+      .catch(err => console.error(err));
+  }
+
+  const handleMessageNewChange = (e) => {
+    setMessageNew(e.target.value);
+  }
+
+  const handleFormSubmit = (e) => {
+    e.preventDefault();
+
+    const options = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({ message: messageNew })
+    }
+
+    fetch(HAPPY_THOUGHTS_URL, options)
+      .then(res => res.json())
+      .then(recievedMessage => setMessageList([...messageList, recievedMessage]))
+      .catch(err => console.error(err));
+
+// set empty string here. Update value of a state property
+      e.target.reset()
+  }
+
+  const handleLikesIncrease = (id) => {
+    const options = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+    };
+
+    fetch(HEARTS_URL(id), options)
+      .then(res => res.json())
+      //fetch data
+      .then(recievedMessage => {
+        const updatedMessageList = messageList.map(input => {
+          if (input._id === recievedMessage._id) {
+            input.hearts += 1;
+          } 
+          return input;
+        });
+        setMessageList(updatedMessageList);
+      })
```

Comment:
You can do the same here 😊 as for the handleFormSubmit. Instead of the long logic for updating the hearts increase you can at the same time POST Request and GET request for the new data. Of course it will be a little bit slower showing the amount of hearts. Try it out if you would like to! ❤️